src/shims/foreign_items.rs

```diff
@@ -227,14 +223,13 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         let (dest, ret) = match ret {
             None => match link_name {
                 "miri_start_panic" => {
-                    this.check_abi(abi, Abi::Rust)?;
-                    this.handle_miri_start_panic(args, unwind)?;
+                    this.handle_miri_start_panic(abi, link_name_sym, args, unwind)?;
                     return Ok(None);
                 }
                 // This matches calls to the foreign item `panic_impl`.
                 // The implementation is provided by the function with the `#[panic_handler]` attribute.
                 "panic_impl" => {
-                    this.check_abi(abi, Abi::Rust)?;
+                    this.check_abi_and_shim_symbol_clash(abi, Abi::Rust, link_name_sym)?;
```

Comment:
Please explain why we don't `check_shim` here (I guess wrong argument count will become obvious when the returned MIR body's stack frame is being set up)